### PR TITLE
Added Minesweeper implementation in C++

### LINKED ISSUE #1598: Minesweeper algorithm implementation in C++
I would like to make a PR on the implementation of minesweeper algorithm in C++ for GSSoC'20.

### PR DESCRIPTION
**Pull request by chandrikadeb7**
Fixes #1598 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added minesweeper algo implementation in C++
- 

#### Languages Used:
- C++
- 

#### Files Added:
- Minesweeper/minesweeper.cpp
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by chandrikadeb7** [comment]
@raghav-dalmia Review this soon Sir.

File: Minesweeper/minesweeper.cpp [hunk 1]
```diff
@@ -0,0 +1,408 @@
+/*
```
Comment (jainaman224) [inline Minesweeper/minesweeper.cpp]: Remove this.

File: Minesweeper/minesweeper.cpp [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+int MINES;
+
+void clrscr()
+{
+    std::cout << "\33[2J\33[1;1H" << std::flush;
+}
+
+bool isvalid(int row,int col)
+{
+   return (row>=0)&&(row<SIDE)&&(col>=0)&&(col<SIDE);
```
Comment (jainaman224) [inline Minesweeper/minesweeper.cpp]: Spaces around operators like `&&`, `>=`, `+`, `-`.

File: Minesweeper/minesweeper.cpp [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+}
+
+bool ismine(int row,int col,char board[][max_side])
+{
+    if(board[row][col]=='B')
+	    return (true);
+    else
+	    return (false);
+}
+void move(int *x,int *y)
```
Comment (jainaman224) [inline Minesweeper/minesweeper.cpp]: New line above function.

File: Minesweeper/minesweeper.cpp [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+	    return (true);
+	}
+	else
+	{
+	    int count=countadjacent(row,col,mines,realboard);
+	    (*moves_left)--;
+	    myboard[row][col]= count+'0';
+	    if(!count)
+	    {
+		    if(isvalid(row-1,col)==true)
```
Comment (jainaman224) [inline Minesweeper/minesweeper.cpp]: Incorrect indentaton. Use 4 space indentation.

File: Minesweeper/minesweeper.cpp [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+
+bool isvalid(int row,int col)
+{
+   return (row >= 0) && (row < SIDE) && (col >= 0) && (col < SIDE);
+}
+
+bool ismine(int row,int col,char board[][max_side])
+{
+    if(board[row][col]=='B')
+		return (true);
```
Comment (jainaman224) [inline Minesweeper/minesweeper.cpp]: Incorrect indentation.

File: Minesweeper/minesweeper.cpp [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+    }
+    return;
+}
+
+int countadjacent(int row,int col,int mines[][2],char realboard[][max_side])
+{
+    int count=0;
+    if(isvalid(row-1,col)==true)
+    {
+    	if(ismine(row-1,col,realboard)==true)
```
Comment (jainaman224) [inline Minesweeper/minesweeper.cpp]: Inconsistent indentation.

File: Minesweeper/minesweeper.cpp [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+    {
+		if(ismine(row,col-1,realboard)==true)
+			count++;
+    }
+    if(isvalid(row-1,col-1)==true)
+    {
+		if(ismine(row-1,col-1,realboard)==true)
+	        count++;
+    }
+    if(isvalid(row-1,col+1)==true)
```
Comment (jainaman224) [inline Minesweeper/minesweeper.cpp]: Spaces around operators like `==`, `+`.

File: Minesweeper/minesweeper.cpp [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+    {
+		if(ismine(row-1,col+1,realboard)==true)
```
Comment (jainaman224) [inline Minesweeper/minesweeper.cpp]: Space after operator like `,`.

File: Minesweeper/minesweeper.cpp [hunk 9]
```diff
@@ -0,0 +1,413 @@
...
+    printf("\nEnter your move as (row col) -> ");
+    scanf("%d %d", x, y);
+    return;
+}
+
+void board(char myboard[][max_side])
+{
+    clrscr();
+    printf("\n\n\t\t\t    ");
+    for(int i=0;i<SIDE;i++)
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: Add space around the operator.

File: Minesweeper/minesweeper.cpp [hunk 10]
```diff
@@ -0,0 +1,413 @@
...
+    {
+        if (i>9)
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: Add space around the operator.

File: Minesweeper/minesweeper.cpp [hunk 11]
```diff
@@ -0,0 +1,413 @@
...
+            printf("%d ",i/10);
+	    else
+	        printf("  ");
+    }		
+    printf("\n\t\t\t    ");
+    for(int i=0;i<SIDE;i++)
+	    printf("%d ",i%10);
+    printf("\n\n");
+    for(int i=0;i<SIDE;i++)
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: Add space around the operator.

File: Minesweeper/minesweeper.cpp [hunk 12]
```diff
@@ -0,0 +1,413 @@
...
+	    MINES=10;
+    }
+    else if(ch == 1)
+	{
+		SIDE=16; 
+	  	MINES=40;
+	}
+	else if(ch == 2)
+	{
+		SIDE=24;
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: Add space around the operator.

**Comment by somya-kapoor** [comment]
Add space around the operator like==,=,<<,>>,+,etc

File: Minesweeper/minesweeper.cpp [hunk 13]
```diff
@@ -0,0 +1,412 @@
...
+}
+
+void board(char myboard[][max_side])
+{
+    clrscr();
+    printf("\n\n\t\t\t    ");
+    for(int i = 0; i < SIDE; i++)
+    {
+        if (i > 9)
+            printf("%d ",i/10);
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: Add space around the operator

File: Minesweeper/minesweeper.cpp [hunk 14]
```diff
@@ -0,0 +1,412 @@
...
+        else
+            printf("  ");
+    }
+    printf("\n\t\t\t    ");
+    for(int i = 0; i < SIDE; i++)
+        printf("%d ",i%10);
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: Add space around the operator

File: Minesweeper/minesweeper.cpp [hunk 15]
```diff
@@ -0,0 +1,412 @@
...
+    printf("\n\n");
+    for(int i = 0; i < SIDE; i++)
+    {
+        printf("\t\t\t    ");
+        for(int j = 0; j < SIDE; j++)
+        {
+            printf("%c ",myboard[i][j]);
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: Add a space after ,

File: Minesweeper/minesweeper.cpp [hunk 16]
```diff
@@ -0,0 +1,412 @@
...
+        }
+        printf(" %2d",i);
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: Add a space after ,

File: Minesweeper/minesweeper.cpp [hunk 17]
```diff
@@ -0,0 +1,412 @@
...
+    }
+    return;
+}
+
+int countadjacent(int row, int col, int mines[][2], char realboard[][max_side])
+{
+    int count = 0;
+    if(isvalid(row - 1, col) == true)
+    {
+    	if(ismine(row - 1, col, realboard) == true)
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: inconsistency in indentation

File: Minesweeper/minesweeper.cpp [hunk 18]
```diff
@@ -0,0 +1,412 @@
...
+
+bool playuntil(char myboard[][max_side], char realboard[][max_side], int mines[][2], int row, int col, int *moves_left)
+{
+	if(myboard[row][col] != '-')
+	    return false;
+	int i, j;
+	if(realboard[row][col] == 'B')
+	{
+	    myboard[row][col] = 'B';
+	    for(i = 0;i < MINES; i++)
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: add space after ;

File: Minesweeper/minesweeper.cpp [hunk 19]
```diff
@@ -0,0 +1,412 @@
...
+    }
+}
+
+void placemines(int mines[][2],char realboard[][max_side])
+{
+    bool mark[max_side*max_side];
+    memset(mark, false, sizeof(mark));
+    for(int i = 0; i < MINES;)
+    {
+        int random = rand() % (SIDE*SIDE);
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: add space around the operator

File: Minesweeper/minesweeper.cpp [hunk 20]
```diff
@@ -0,0 +1,412 @@
...
+        int x = random/SIDE;
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: add space around the operator

File: Minesweeper/minesweeper.cpp [hunk 21]
```diff
@@ -0,0 +1,412 @@
...
+        int y = random%SIDE;
```
Comment (somya-kapoor) [inline Minesweeper/minesweeper.cpp]: add space around the operator

**Comment by raghav-dalmia** [comment]
Squash your commits. Rest looks good to me.

**Pull request closed without merging**